### ISSUE TITLE
Part annotations placement is wrong

### ISSUE THREAD

**Issue by riazanovskiy** [trevorsandy/lpub3d#267]
### Subject
Part annotation placement is lower than needed

### Environment
LPub3D 2.3.11
Windows 10

### Configuration
Same as in #261 (please ping me if any new configuration details are necessary)

### Screenshots
![image](https://user-images.githubusercontent.com/11013917/58192249-f65b6900-7cc8-11e9-85d3-bbf3212ee013.png)

```
0 Author: LDraw
0 !LPUB PLI ANNOTATION DISPLAY GLOBAL TRUE
1 1 -70 -1 10 1 0 0 0 1 0 0 0 1 87080.dat
1 4 -150 -1 10 1 0 0 0 1 0 0 0 1 87086.dat
1 0 -110 -41 50 0 1 0 -1 0 0 0 0 1 32278.dat
1 0 -80 -41 30 1 0 0 0 1 0 0 0 1 3706.dat
1 71 -10 -41 30 0 0 1 0 1 0 -1 0 0 32016.dat
```

**Comment by trevorsandy** [comment]
Thanks for reporting this behaviour. Unfortunately, I cannot reproduce the output you have shown. 

I downloaded and extracted a portable distribution of LPub3D v2.3.11 and opened the model example you provided. Here are my results:

![Screenshot - 22_05_2019 , 20_00_14](https://user-images.githubusercontent.com/13388970/58197526-c01fe880-7ccc-11e9-954b-38904d8e59d4.png)

![Screenshot - 22_05_2019 , 20_01_01](https://user-images.githubusercontent.com/13388970/58197527-c01fe880-7ccc-11e9-8d4b-b0805e011191.png)

Cheers,

**Comment by riazanovskiy** [comment]
I have tried switching renderer between LDView and Native (I am using LDView because the native one fails on my instruction) and swithing units between inches and centimeters.
I also tried reinstalling LPub, carefully removing all config files and registry keys. All this to effect. I will try portable version.

I also tried installing LPub on my Arch linux (from github releases) and with default config the annotation placement is also not entirely correct:
![Screenshot_2019-05-23_14-43-41](https://user-images.githubusercontent.com/11013917/58250060-3114de00-7d69-11e9-96db-16ff037e936c.png)

Settings screenshots:
![Screenshot_2019-05-23_14-49-53](https://user-images.githubusercontent.com/11013917/58250429-14c57100-7d6a-11e9-94a1-938f5bcb2680.png)
![Screenshot_2019-05-23_14-49-42](https://user-images.githubusercontent.com/11013917/58250430-14c57100-7d6a-11e9-89ac-01fcee4d8631.png)


Random suggestion: can this have anything to do with system DPI setting?

**Comment by riazanovskiy** [comment]
(@trevorsandy please note that I cannot reopen an issue closed by a repository collaborator)

Yes, in fact, the vertical placement of annotation text relative to annotation box does depend on system DPI setting! I tried changing it on Linux and the text position changed.
This would also explain why it differs with my Windows setup, they have different system DPI setting

**Comment by trevorsandy** [comment]
Very well – what are your DPI settings - available under Project Globals Setup ?

When you change most settings, a “clear and reload model file components - temp, parts, assemblies etc...” is initiated allowing LPub3D to properly map an annotation rect to its annotation style rect during composition of the PLI.

You can also “clear and redraw” the model manually from the LDraw Editor menu.

I’ll take a look at the impact of the DPI setting on the rects relationship.

Cheers,

**Comment by trevorsandy** [comment]
> I am using LDView because the native one fails on my instruction

Can I reproduce this failure ? You can provide the LPub3D log file.

Cheers,

**Comment by riazanovskiy** [comment]
Did you mean this screen? It's 381 there. I have never changed that
![Screenshot_2019-05-23_16-27-52](https://user-images.githubusercontent.com/11013917/58256537-db93fd80-7d77-11e9-964e-c8442a2d5f5d.png)

Just in case I wasn't clear, I meant system DPI setting, not LPub one (in my case of XFCE it is System settings -> Appearance -> Fonts -> Custom DPI setting: 140). Windows also has similar setting

**Comment by riazanovskiy** [comment]
> > I am using LDView because the native one fails on my instruction
> 
> Can I reproduce this failure ? You can provide the LPub3D log file.
> 

Surely, I'll create another issue. I hesitated to do it because it fails on my huge (12 MB) instruction file and I have not yet created a minimal example

**Comment by trevorsandy** [comment]
> Did you mean this screen? It's 381 there. I have never changed that

I meant whichever DPI setting you are manipulating that alters the annotation display to show the expected and unexpected behaviour.

So if I well understand, your system font custom DIP setting on Arch is affecting the annotation display ?

To save time, what are your Windows platform system DIP settings that produce the unexpected behaviour and are you able to alter those settings to correct the annotation display ?

Cheers,

**Comment by riazanovskiy** [comment]
Yes, you're right, that is the XFCE setting which affects annotation text placement.

I checked it under Windows, it works the same. If I set 100% in Settings (Windows 10 app) -> Display -> Scale and layout, the annotation text is placed correctly. If I enter anything bigger (I had 175%), the text moves down

**Comment by trevorsandy** [comment]
I was able to get the font to scale by multiplying the floating point size by a factor (divide the style rect by the text rect). However, I can’t seem to get the style rect to automatically scale. I think I’ll just have to capture the system DPI scale factor and do the rect scaling manually.

Cheers

**Comment by trevorsandy** [comment]
Implemented.

I compute the rect scale ratio by dividing the system DPI against the constant of 96 DPI. 

The following sample shots show the Windows behaviour; However, I don't expect different behaviour on X11 or macOS.

![Screenshot - 24_05_2019 , 20_48_18](https://user-images.githubusercontent.com/13388970/58350439-2db34c80-7e66-11e9-95dc-ccffba70e3b5.png)

![Screenshot - 24_05_2019 , 20_46_51](https://user-images.githubusercontent.com/13388970/58350444-32780080-7e66-11e9-9682-ab6f44a1f47d.png)

I noticed there are several other fixed-size item that will have to be manually scaled - notably the page background, and CSI image. The scene ruler should also be adjusted. So there is more to be done but I'll add those updates on separate tickets.

Cheers,
